corehq/apps/sms/phonenumbers_helper.py

```diff
@@ -17,23 +17,46 @@ def parse_phone_number(number, region=None, failhard=True):
     from phonenumbers.phonenumberutil import parse as phonenumbers_parse, NumberParseException
     try:
         phone = phonenumbers_parse(number, region)
-        if phone:
-            return PhoneNumber(phone.country_code, phone.national_number)
     except NumberParseException:
         if failhard:
             raise PhoneNumberParseException()
         else:
             return None
 
+    return PhoneNumber(phone.country_code, phone.national_number)
+
 
 def strip_plus(number):
     return number[1:] if number.startswith('+') else number
 
 
-def get_country_code_and_national_number(number, failhard=False):
-    parsed = parse_phone_number(number, failhard=failhard)
-    if parsed:
-        country_code = parsed.country_code
-        national_number = strip_plus(number)[len(str(country_code)):]
-        return country_code, national_number
-    return None, None
+def get_country_code_and_national_number(number):
+    """
+    :param number: str representing phone number
+    :return: country_code: int, national_number: str or, if the number
+    could not be parsed, `None, None`.
+    """
+    parsed = parse_phone_number(number, failhard=False)
+    return (parsed.country_code, str(parsed.national_number)) if parsed else (None, None)
+
+
+def country_code_for_region(region):
+    """
+    Simple wrapper around phonenumbers country_code_for_region method
+    :param region: str (examples: 'US', 'IN', 'ZA')
+    :return: int, 0 if invalid, country_code if valid
+    NOTE it is possible for this to raise an Exception if phonenumbers region validation has a bug
```

Comment:
They raise a general `Exception` which is certainly surprising, but maybe that is because from their perspective, it should _never_ happen. 

We can't add a test to exercise this code because it is all internal to phonenumbers. They have a list of supported regions to check if a region is valid, and then they have a method to retrieve `PhoneMetadata` for a region that appears distinct from the list of supported regions, hence the chance of an exception.

Based on your suggestion to change from if to because, I'm wondering if we are on slightly different pages and if this comment clears anything up. I wouldn't say that there is a bug in phonenumbers at the moment.

Reply:
> They raise a general `Exception`

That is terrible, and either an indication of someone not knowing how to use exceptions or it being an exception that is not meant to be caught (which would be surprising).

> We can't add a test to exercise this code because it is all internal to phonenumbers. ... I wouldn't say that there is a bug in phonenumbers at the moment.

I think I'm not understanding. It seems like there would be specific values, if passed to `country_code_for_region` will cause it to raise an exception. Can we not find such a value and write a test that passes that value into `country_code_for_region` and assert that the exception is raised?

Reply:
Their expected behavior is to determine that a region is invalid before attempting to use it, and return 0 as the country code (not raise an exception). I have a test for this behavior (see `test_county_code_for_region_returns_zero_if_invalid_region`).

Reply:
Offlined with Daniel and it turned out to be some miscommunication. Given that the raised exception is only possible if there is a bug in phonenumbers, we are not going to document it on our side. Removed in 9a16ed9eaeb9a946982ab01180175c3ef8637e74.